Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -188,8 +188,8 @@ public OrderTicket MarketOrder(Symbol symbol, int quantity, bool asynchronous =
             //Add the order and create a new order Id.
             var ticket = Transactions.AddOrder(request);
 
-            //Wait for the order event to process, only if the exchange is open
-            if (!asynchronous)
+            // In live mode, wait for the order event to process, only if the exchange is open
+            if (!asynchronous && LiveMode)
```

Comment:
We actually allow synchronous orders to fill in backtest as well. This is so people can say "MarketOrder()" ... and then have the portfolio filled on the next line of code. (This is what the 5 second timeout was for -- let people have a synchronous order which times out and continues)

I understand this will make delayed fill difficult :) So to keep this behavior -- the delayed time step should probably only apply to non-market orders?